### ISSUE TITLE
False positive GH_TOKEN vs GH_ENTERPRISE_TOKEN

### ISSUE THREAD

**Issue by TWiStErRob** [TWiStErRob/net.twisterrob.ghlint#246]
### Version

0.5.0

### Reproduction steps

1. File `.github/workflows/x.yml` with the following content:
   ```yaml
   on: push
   jobs:
     repro:
       runs-on: ubuntu-latest
       steps:
         - run: gh pr list
           env:
             GH_ENTERPRISE_TOKEN: ${{ github.token }}
             GH_HOST: https://github.my-company.com/
   ```

2. Run `ghlint` with the following arguments:
   ```shell
   ghlint --console .github/workflows/x.yml
   ```

### Expected behavior

No finding, setup is correct.

### Actual behavior

Flagged because GH_TOKEN is missing.

### Additional Context

https://cli.github.com/manual/gh_help_environment

Note: also write an issue in the same rule for missing `GH_HOST`, and/or create a new ticket.
